piff/polynomial_interp.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: interp_mean
+"""
+
+from __future__ import print_function
+
+from .interp import Interp
+import numpy
+from numpy.polynomial.polynomial import polyval2d
+from numpy.polynomial.chebyshev import chebval2d
+from numpy.polynomial.legendre import legval2d
+from numpy.polynomial.laguerre import lagval2d
+from numpy.polynomial.hermite import hermval2d
+
+POLYNOMIAL_TYPES = {
+    "POLY":polyval2d,
+    "CHEBYSHEV":chebval2d,
+    "LEGENDRE":legval2d,
+    "LAGUERRE":lagval2d,
+    "HERMITE":hermval2d,
+}
+
+
+class Polynomial(Interp):
+    """
+    An interpolator that uses  scipy curve_fit command to fit a polynomial 
+    surface to each parameter passed in independently.
+
+    :param orders:  List/array of integers, one for each parameter 
+                    to be interpolated.
+    """
+    def __init__(self, orders, poly_type="POLY"):
+        """Create 
+        """
+        self.set_orders(orders)
+        self.set_function(poly_type)
+        self.coeffs = None
+
+    def set_orders(self, orders):
```

Comment:
Likewise for some of the other methods below.  `set_function`, `generate_indices`, `pack_coefficients`, etc.